packages/monaco/src/browser/monaco-outline-contribution.ts

```diff
@@ -32,27 +32,28 @@ import debounce = require('lodash.debounce');
 @injectable()
 export class MonacoOutlineContribution implements FrontendApplicationContribution {
 
-    protected ids: string[] = [];
-    protected symbolList: NodeAndSymbol[] = [];
-    protected readonly outlineSymbolInformations: MonacoOutlineSymbolInformationNode[];
     protected cancellationSource: CancellationTokenSource;
+    protected readonly toDisposeOnClose = new DisposableCollection();
+    protected readonly toDisposeOnEditor = new DisposableCollection();
 
-    constructor(
-        @inject(OutlineViewService) protected readonly outlineViewService: OutlineViewService,
-        @inject(EditorManager) protected readonly editorManager: EditorManager) { }
+    @inject(OutlineViewService) protected readonly outlineViewService: OutlineViewService;
+    @inject(EditorManager) protected readonly editorManager: EditorManager;
 
     onStart(app: FrontendApplication): void {
-        this.outlineViewService.onDidChangeOpenState(async isOpen => {
-            this.updateOutline();
+        this.outlineViewService.onDidChangeOpenState(async open => {
+            if (open) {
+                this.toDisposeOnClose.push(this.toDisposeOnEditor);
+                this.toDisposeOnClose.push(DocumentSymbolProviderRegistry.onDidChange(
+                    debounce(() => this.updateOutline())
```

Comment:
@akosyakov, I'm confused that there is no delay here. AFAIK `DocumentSymbolProviderRegistry` might flood us here with events. I think that debounce should have a delay of some ms and also it should be called with `{ leading: true }`, so that we start seeing something right after opening the editor.

Reply:
I have not changed it, it was like that before. Could you check and file an issue if it is true?